components/SearchBar.jsx

```diff
@@ -1,29 +1,97 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect, useRef } from 'react';
 //tools
 import { DateRange } from 'react-date-range';
-
+import format from 'date-fns/format';
+import { useRouter } from 'next/router';
 //styles
-import 'react-date-range/dist/styles.css'; // main style file
-import 'react-date-range/dist/theme/default.css'; // theme css file
-import { MagnifyingGlassIcon, UsersIcon } from '@heroicons/react/24/solid';
+import {
+	MagnifyingGlassIcon,
+	UsersIcon,
+	PlusCircleIcon,
+	MinusCircleIcon,
+} from '@heroicons/react/24/solid';
 
 function SearchBar() {
 	//search
 	const [searchInput, setSearchInput] = useState('');
+	const router = useRouter();
+	const search = (event) => {
+		event.preventDefault();
+		if (searchInput !== '') {
+			router.push({
+				pathname: '/search',
+				query: {
+					location: searchInput !== '' ? searchInput : '',
+					startDate: startDate.toISOString(),
+					endDate: endDate.toISOString(),
+					noOfGuests,
+					servicesSelected,
+				},
+			});
+
+			return setSearchInput('');
+		} else if (searchInput === '') {
+			router.push({
+				pathname: '/search/all',
+				query: {},
+			});
+			return setSearchInput('');
+		} else {
+			return alert('please enter a location');
```

Comment:
Why do you return the undefined of the alert? 
```suggestion 
			alert('please enter a location');
			return
```